llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -394,11 +394,12 @@ bool PreISelIntrinsicLowering::expandMemIntrinsicUses(Function &F) const {
       Module *M = Memset->getModule();
       const DataLayout &DL = Memset->getDataLayout();
 
+      Type *DestPtrTy = Memset->getRawDest()->getType();
+      Type *IntIdxTy = DL.getIndexType(DestPtrTy);
```

Comment:
As we're emitting a libcall here, it would be preferable to use TLI.getSizeTType() instead. Can also use the getAsSizeT helper for the ConstantInt below.

It will ultimately produce the same type, but this way we're not baking in new assumptions.

Reply:
@asb - This doesn't appear to have been addressed.

Reply:
Apologies for missing this, now updated.